libarchive/test/test_pax_filename_encoding.c

```diff
@@ -650,11 +651,63 @@ DEFINE_TEST(test_pax_filename_encoding_UTF16_win)
 	archive_entry_free(entry);
 	assertEqualInt(ARCHIVE_OK, archive_write_free(a));
 
-	/* Ensure that the names round trip properly */
+	/*
+	 * Examine the bytes to ensure the filenames ended up UTF-8
+	 * encoded as we expect.
+	 */
```

Comment:
This new block of test code verifies that the write operation above really did create a file with pax extensions, as expected.